docs/reference/numbers.tex

```diff
@@ -44,16 +44,18 @@
 
 // \ref{num.assoc.types}, associated types
 template<typename T>
-struct number_scalar;
+struct vector_scalar;
```

Comment:
So we have `scalar_quantity` that is a vector, and we can get its `vector_scalar`? And we will also get `vector_quantity` and `tensor_quantity` in the future 😉 I am afraid that is still confusing. 

It is also specialized for `std::complex`, which is not a vector.

Reply:
Should I rename it to `scalar_of_vector_space`?

Reply:
Renaming it also made it inconsistent with `number_difference_t`.
Maybe I should rename that to `point_difference_t`.
![1695509969](https://github.com/mpusz/mp-units/assets/21071787/b59ad722-708c-4850-9068-be64e04cfbf8)

Reply:
> So we have `scalar_quantity` that is a vector, and we can get its `vector_scalar`? And we will also get `vector_quantity` and `tensor_quantity` in the future 😉 I am afraid that is still confusing.

I don't know.
I provided `scalar_quantity` (from <https://www.electropedia.org/iev/iev.nsf/display?openform&ievref=102-02-19>).
I used it once to constrain a Cartesian vector of quantities.
But then reworked it to have similar type names:
`cartesian_vector2d<quantity<metre>>` -> `cartesian_vector2d<metre>`.

Now it's mostly a placeholder for the part of the reference documentations for mp-units.